spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitTemplateObservation.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.support.micrometer;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.Observation.Context;
+import io.micrometer.observation.ObservationConvention;
+import io.micrometer.observation.docs.DocumentedObservation;
+
+/**
+ * Spring RabbitMQ Observation for {@link org.springframework.amqp.rabbit.core.RabbitTemplate}.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public enum RabbitTemplateObservation implements DocumentedObservation {
```

Comment:
You know, in Spring Integration I'm planning to have all the observation impls in one class.
WDYT?

Reply:
For what purpose?

Reply:
Just to have all the observations in one place. The `enum` is not restricted only to a single entry...
I'm not sure how the final doc is going to look, but that's my thought: have all the built-in observations in one place.

@marcingrzejszczak , what is your idea on the matter, please?

Reply:
You don't need to have them in one class, we will do the scan of all of your sources and merge the results together (https://micrometer.io/docs/tracing#_documentation_building)